cpp/include/cudf/column/column_view.hpp

```diff
@@ -375,6 +380,51 @@ class column_view : public detail::column_view_base {
    */
   auto child_end() const noexcept { return _children.cend(); }
 
+  /**
+   * @brief Construct a column view from a device_span<T>.
+   *
+   * Only numeric and chrono types are supported.
+   *
+   * @tparam T The device_span data type.
+   * @param data The device_span containing the column elements.
+   */
+  template <typename T, std::enable_if_t<cudf::is_numeric<T>() or cudf::is_chrono<T>()>* = nullptr>
+  [[nodiscard]] column_view(device_span<T> data)
+    : column_view(cudf::data_type{cudf::type_to_id<std::remove_const<T>>()},
+                  data.size(),
+                  data.data(),
+                  nullptr,
+                  0,
+                  0,
+                  {})
+  {
+    CUDF_EXPECTS(data.size() < std::numeric_limits<cudf::size_type>::max(),
+                 "Device span exceeds maximum size of a column view.");
+  }
+
+  /**
+   * @brief Converts a column view into a device span.
+   *
+   * Only numeric and chrono types are supported.
+   *
+   * @tparam The device_span type. Must be const and match the column view's type.
+   *
+   * @throws cudf::logic_error if the column view has nulls.
+   * @throws cudf::logic_error if the column view type does not match the span type.
+   *
+   * @return device_span<T> A typed device span of the column view.
+   */
+  template <typename T,
+            typename std::enable_if_t<std::is_const_v<T> &&
+                                      (cudf::is_numeric<T>() || cudf::is_chrono<T>())>* = nullptr>
+  [[nodiscard]] operator device_span<T>() const
```

Comment:
What's the reason for putting an `is_const` check in the enable if instead of `operator device_span<T const>`?

Reply:
I wasn't 100% sure how the type deduction worked so I tried the `is_const` check, but it turned out I had a different problem and `T const` works fine -- I already changed this locally but haven't pushed yet. This PR is still a draft. 😉